conans/test/unittests/util/tools_test.py

```diff
@@ -794,6 +795,30 @@ def vswhere_description_strip_test(self):
             json = vswhere()
             self.assertNotIn("descripton", json)
 
+    @unittest.skipUnless(platform.system() == "Windows", "Requires vswhere")
+    def vswhere_path_test(self):
+        # vswhere not found
+        with tools.environment_append({"ProgramFiles": None, "ProgramFiles(x86)": None, "PATH": ""}):
+            with self.assertRaisesRegex(ConanException, "Cannot locate vswhere"):
+                vswhere()
+        # vswhere in ProgramFiles
+        program_files = get_env("ProgramFiles(x86)") or get_env("ProgramFiles")
+        vswhere_path = None
+        if program_files:
+            expected_path = os.path.join(program_files, "Microsoft Visual Studio", "Installer",
+                                         "vswhere.exe")
+            if os.path.isfile(expected_path):
+                vswhere_path = expected_path
+                with tools.environment_append({"PATH": ""}):
+                    self.assertTrue(vswhere())
+        # vswhere in PATH
+        env = {"ProgramFiles": None, "ProgramFiles(x86)": None}
+        if not which("vswhere") and vswhere_path:
+                vswhere_folder = os.path.join(program_files, "Microsoft Visual Studio", "Installer")
+                env.update({"PATH": [vswhere_folder]})
```

Comment:
Is it not something we want to do, even if vswhere is in the path, and which() returned True?

Reply:
The rationale is the following: If we know (from previous case) where is vswhere and we dont found it in the PATH (which() returned None). Then set vswhere_path value in the path so it can be found only there (that is the previus conditions of the test: "vswhere in PATH but not in ProgramFiles")